TESTING.md

```diff
@@ -47,6 +47,17 @@ Bugs fixes should include a unit test case which exercises the bug.
 A bug fix may also include new assertions in an existing integration tests for the
 API endpoint.
 
+### Integration tests environment considerations
+
+When adding new tests or modifying existing test under `integration/`, testing environment should be properly considered. `skip.If` from [gotestyourself/skip](https://godoc.org/gotest.tools/skip) can be used to make the test run conditionally. Full testing environment conditions can be found at [environment.go](https://github.com/moby/moby/blob/master/internal/test/environment/environment.go)
+
+Here is a quick example. If the test needs to interact with a docker daemon on the same host, the following condition should be checked within the test code
+```go
```

Comment:
Can you add a blank line before and after this code-block? Markdown sometimes is picky about that

Reply:
Done.